packages/fast-check/src/arbitrary/_internals/implementations/SchedulerImplem.ts

```diff
@@ -184,12 +184,16 @@ export class SchedulerImplem<TMetaData> implements Scheduler<TMetaData> {
           customAct || defaultSchedulerAct,
           () => builder(),
         );
-        scheduled.catch(onFaulty);
         return scheduled;
       };
-      previouslyScheduled = previouslyScheduled.then(onNext);
+      // We will run current item if and only if the one preceeding it succeeds
+      // Otherwise, we mark the run as "failed" and ignore any subsequent item (including this one)
+      previouslyScheduled = previouslyScheduled.then(onNextItem, onFaultyItem);
     }
-    previouslyScheduled.then(onDone, onFaultyNoop);
+    // Once last item is done, the full sequence can be considered as successful
+    // If it failed (or any preceeding one failed), then the sequence should be marked as failed (already marked for others)
+    // We always handle Promise rejection of previouslyScheduled internally, in other words no error will bubble outside
+    previouslyScheduled.then(onDone, onFaultyItemNoThrow);
```

Comment:
The `then` explained:

- success: last item done! it's a success for the whole sequence: all items passed
- failure:
  - last item failed: we have to mark the sequence as failed
  - another one (throw): we ignore the error (it has already been marked and resolved)